File: 3.FilesIO/Go/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"time"
)

func main() {
	filename := "../input.txt"

	// Start measuring time
	start := time.Now()

	// Open the file
	file, err := os.Open(filename)
	if err != nil {
		log.Fatalf("Error opening file: %v", err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		fmt.Printf(scanner.Text() + "\n")
	}

	elapsed := time.Since(start)

	if err := scanner.Err(); err != nil {
		log.Fatalf("Error reading file: %v", err)
	}

	fmt.Printf("Execution time: %.6f seconds\n", elapsed.Seconds())
}

File: 3.FilesIO/C/main.c
#include <stdio.h>
#include <time.h>

int main() {
    FILE* file;
    const char* filename = "../input.txt";
    char buffer[100];
    clock_t start, end;
    double cpu_time_used;

    start = clock();
    file = fopen(filename, "r");
    if (file == NULL)
    {
        printf("Error opening file.\n");
        return 1;
    }

    while (fgets(buffer, sizeof(buffer), file) != NULL)
    {
        printf("%s", buffer);
    }

    end = clock();
    fclose(file);

    cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
    printf("Execution time: %.6f seconds.\n", cpu_time_used);

    return 0;
}